crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -157,6 +158,48 @@ pub(crate) trait StateWriteExt: StateWrite {
         self.put_raw(nonce_storage_key(address), bytes);
         Ok(())
     }
+
+    #[instrument(skip(self))]
+    async fn increase_balance(
+        &mut self,
+        address: Address,
+        asset: asset::Id,
+        amount: u128,
+    ) -> Result<()> {
+        let balance = self
+            .get_account_balance(address, asset)
+            .await
+            .context("failed to get account balance")?;
+        self.put_account_balance(
+            address,
+            asset,
+            balance
+                .checked_add(amount)
+                .ok_or(anyhow!("account balance overflow"))?,
+        )?;
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    async fn decrease_balance(
+        &mut self,
+        address: Address,
+        asset: asset::Id,
+        amount: u128,
+    ) -> Result<()> {
+        let balance = self
+            .get_account_balance(address, asset)
+            .await
+            .context("failed to get account balance")?;
+        self.put_account_balance(
+            address,
+            asset,
+            balance
+                .checked_sub(amount)
+                .ok_or(anyhow!("account balance underflow; insufficient funds?"))?,
```

Comment:
`ok_or(anyhow!(...))` is not necessary for anyhow, as it impls `Context for Option<T>`:

```suggestion
                .context("subtracting from account balance failed due to insufficient funds")?,
```